pipeline.py

```diff
@@ -51,19 +61,26 @@ class CorpusManager:
     """
 
     def __init__(self, path_to_raw_txt_data: str):
-        pass
+        self._storage = {}
+        self.path_to_raw_txt_data = path_to_raw_txt_data
+        self._scan_dataset()
 
     def _scan_dataset(self):
         """
         Register each dataset entry
         """
-        pass
+        paths_to_files = list(Path(self.path_to_raw_txt_data).glob('*_raw.txt'))
+        id_template = re.compile(r'\d+')
+        for file_path in paths_to_files:
+            file_id = int(re.search(id_template, str(file_path.name)).group())
```

Comment:
as soon as you have a compiled pattern, you need to call methods from it, like: `id_template.search`